projects/plugins/jetpack/extensions/shared/external-media/sources/google-photos/index.js

```diff
@@ -15,23 +16,50 @@ function GooglePhotos( props ) {
 		createPickerSession,
 		fetchPickerSession,
 		getPickerStatus,
+		setAuthenticated,
 	} = props;
 	const [ cachedSessionId ] = useState( getGooglePhotosPickerCachedSessionId() );
 	const [ isCachedSessionChecked, setIsCachedSessionChecked ] = useState( false );
 	const [ pickerFeatureEnabled, setPickerFeatureEnabled ] = useState( null );
+	const [ authUpgradeRequired, setAuthUpgradeRequired ] = useState( false );
 	const isPickerSessionAccurate = pickerSession !== null && ! ( 'code' in pickerSession );
 	const isSessionExpired =
 		pickerSession?.expireTime && moment( pickerSession.expireTime ).isBefore( new Date() );
 
+	const catchAuthErrors = useCallback(
+		error => {
+			if ( error.code === 'authorization_required' ) {
+				setAuthenticated( false );
+			}
+
+			// If the picker session endpoint returns a 404
+			// the user needs to upgrade their auth
+			if ( error.code === 'rest_not_found' ) {
+				setAuthUpgradeRequired( true );
+			}
+		},
+		[ setAuthenticated, setAuthUpgradeRequired ]
+	);
+
+	useEffect( () => {
+		! isAuthenticated && setAuthUpgradeRequired( false );
+	}, [ isAuthenticated ] );
+
 	useEffect( () => {
 		getPickerStatus().then( feature => {
 			feature && setPickerFeatureEnabled( feature.enabled );
 		} );
 
 		cachedSessionId === null && setIsCachedSessionChecked( true );
```

Comment:
Is this needed since the next line is setting the cache check to true anyway?

Reply:
Based on the [following comment](https://github.com/Automattic/jetpack/pull/40416#discussion_r1868761742) if we're putting fetchPickerSession in the condition then you can ignore this.